Stardrop/Utilities/External/NexusClient.cs

```diff
@@ -56,74 +56,124 @@ public static void SetDisplayWindow(MainWindowViewModel viewModel)
+        internal event EventHandler<ModDownloadProgressEventArgs>? DownloadProgressChanged = null;
+        internal event EventHandler<ModDownloadCompletedEventArgs>? DownloadCompleted = null;
+        internal event EventHandler<ModDownloadFailedEventArgs>? DownloadFailed = null;
+
+        public NexusClient(HttpClient client)
+        {
+            _client = client;
+        }
+
+        /// <summary>
+        /// Calls the /validate endpoint on Nexus Mods' API using the API key stored in this client upon creation.
+        /// If the key is successfully validated, its information is cached in <see cref="Program.settings.NexusDetails"/>.<br/>
+        /// Returns <see langword="false"/> if validation fails.
+        /// </summary>
+        public async Task<bool> ValidateKey()
+        {
+            try
+            {
+                var response = await _client.GetAsync("users/validate");
+                if (!response.IsSuccessStatusCode || response.Content == null)
                 {
-                    if (response.StatusCode != System.Net.HttpStatusCode.OK)
+                    Program.helper.Log($"Call to Nexus Mods failed. HTTP status code: {response.StatusCode}, {response.ReasonPhrase}");
+                    if (response.Content == null)
                     {
-                        Program.helper.Log($"Bad status given from Nexus Mods: {response.StatusCode}");
-                        if (response.Content is not null)
-                        {
-                            Program.helper.Log($"Response from Nexus Mods:\n{await response.Content.ReadAsStringAsync()}");
-                        }
+                        Program.helper.Log($"No response from Nexus Mods!");
                     }
-                    else if (response.Content is null)
+                    else
                     {
-                        Program.helper.Log($"No response from Nexus Mods!");
+                        Program.helper.Log($"Response from Nexus Mods:\n{await response.Content.ReadAsStringAsync()}");
                     }
+                    return false;
+                }
+
+
+                string content = await response.Content.ReadAsStringAsync();
+                Validate validationModel = JsonSerializer.Deserialize<Validate>(content, new JsonSerializerOptions { PropertyNameCaseInsensitive = true })!;
+
+                if (validationModel is null || String.IsNullOrEmpty(validationModel.Message) is false)
+                {
+                    Program.helper.Log($"Unable to validate given API key for Nexus Mods");
+                    Program.helper.Log($"Response from Nexus Mods:\n{content}");
 
-                    wasValidated = false;
+                    return false;
                 }
+
+                if (Program.settings.NexusDetails is null)
+                {
+                    return false;
+                }
+
+                Program.settings.NexusDetails.Username = validationModel.Name;
```

Comment:
This is--I think-- the one place where `NexusClient` is still doing things that affect global state. Something to remember, and a reason not to have more than one `NexusClient` kicking around at a time.